homeassistant/components/home_connect/select.py

```diff
@@ -0,0 +1,280 @@
+        "LaundryCare.Washer.Program.ShirtsBlouses",
+        "LaundryCare.Washer.Program.SportFitness",
+        "LaundryCare.Washer.Program.Towels",
+        "LaundryCare.Washer.Program.WaterProof",
+        "LaundryCare.Washer.Program.PowerSpeed59",
+        "LaundryCare.Washer.Program.Super153045.Super15",
+        "LaundryCare.Washer.Program.Super153045.Super1530",
+        "LaundryCare.Washer.Program.DownDuvet.Duvet",
+        "LaundryCare.Washer.Program.Rinse.RinseSpinDrain",
+        "LaundryCare.Washer.Program.DrumClean",
+        "LaundryCare.WasherDryer.Program.Cotton",
+        "LaundryCare.WasherDryer.Program.Cotton.Eco4060",
+        "LaundryCare.WasherDryer.Program.Mix",
+        "LaundryCare.WasherDryer.Program.EasyCare",
+        "LaundryCare.WasherDryer.Program.WashAndDry60",
+        "LaundryCare.WasherDryer.Program.WashAndDry90",
+    )
+}
+
+PROGRAMS_TRANSLATION_KEYS_MAP = {
+    value: key for key, value in TRANSLATION_KEYS_PROGRAMS_MAP.items()
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Home Connect select entities."""
+
+    def get_entities() -> list[HomeConnectProgramSelectEntity]:
+        """Get a list of entities."""
+        entities: list[HomeConnectProgramSelectEntity] = []
+        hc_api: ConfigEntryAuth = hass.data[DOMAIN][config_entry.entry_id]
+        for device in hc_api.devices:
+            if device.appliance.type in APPLIANCES_WITH_PROGRAMS:
+                with contextlib.suppress(HomeConnectError):
+                    programs = device.appliance.get_programs_available()
+                    if programs:
+                        entities.extend(
+                            HomeConnectProgramSelectEntity(
+                                device, programs, start_on_select
+                            )
+                            for start_on_select in (True, False)
+                        )
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectProgramSelectEntity(HomeConnectEntity, SelectEntity):
+    """Select class for Home Connect programs."""
+
+    def __init__(
+        self, device: HomeConnectDevice, programs: list[str], start_on_select: bool
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(
+            device,
+            SelectEntityDescription(
+                key=BSH_ACTIVE_PROGRAM if start_on_select else BSH_SELECTED_PROGRAM,
+                translation_key="active_program"
+                if start_on_select
+                else "selected_program",
+            ),
+        )
+        self._attr_options = []
+        for program in programs:
+            if program not in PROGRAMS_TRANSLATION_KEYS_MAP:
+                _LOGGER.info(
+                    'The program "%s" is not contemplated on this integration', program
+                )
```

Comment:
@MartinHjelmare Is this log ok? Should we log something different?

Reply:
Something like my suggestion above, I think.

Reply:
We should probably already do this before creating the entity though. So we don't spam the log multiple times with the same message.

Reply:
Once per appliance and program would be okey?

Reply:
Only once per program I think. Ie, we don't need to log more than once for each missing program key. It doesn't matter what appliances has the the program.